commands/Announcement/announcement.js

```diff
@@ -6,20 +6,18 @@ module.exports = class extends Command {
     constructor(...args) {
         super(...args, {
             name: 'announcement',
-            permLevel: 2,
+            permissionLevel: 2,
             runIn: ['text'],
             description: 'Send new announcements, mentioning the announcement role.',
             usage: '<string:string>',
-            cooldown: 60
+            cooldown: 60,
+            requiredSettings: ['channels.announcementChannel', 'roles.announcementRole']
         });
     }
 
     async run(msg, [message]) {
         const announcementID = msg.guild.settings.channels.announcementChannel;
-        if (!announcementID) throw msg.language.get('COMMAND_SUBSCRIBE_NO_CHANNEL');
```

Comment:
Why did you remove the checks?

Reply:
because required permissions will throw if they're not set, I can put them back if you would prefer however

Reply:
Yes, please.

Reply:
Actually, only this check should go. The other two are fine.

Reply:
So for clarification, just re-add the check for announcement channel?

Reply:
And if it's postable. Because that check (requiredPermissions) checks for the value, but not if the resolved value exists.